src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -5,14 +5,1235 @@
+            DataPackage dataPack = DataPackage();
+            dataPack.RequestedOperation(DataPackageOperation::Copy);
+            dataPack.SetText(copiedData);
+            Clipboard::SetContent(dataPack);
+
+            // TODO: MSFT 20642290 and 20642291
+            // rtf copy and html copy
+        });
+    }
+
+    // Method Description:
+    // - Fires an async event to get data from the clipboard, and paste it to
+    //   the terminal. Triggered when the Terminal Control requests clipboard
+    //   data with it's PasteFromClipboard event.
+    // Arguments:
+    // - eventArgs: the PasteFromClipboard event sent from the TermControl
+    void TerminalPage::_PasteFromClipboardHandler(const IInspectable& /*sender*/,
+                                                  const PasteFromClipboardEventArgs& eventArgs)
+    {
+        this->Dispatcher().RunAsync(CoreDispatcherPriority::High, [eventArgs]() {
+            TerminalPage::PasteFromClipboard(eventArgs);
+        });
+    }
+
+    // Function Description:
+    // - Copies and processes the text data from the Windows Clipboard.
+    //   Does some of this in a background thread, as to not hang/crash the UI thread.
+    // Arguments:
+    // - eventArgs: the PasteFromClipboard event sent from the TermControl
+    fire_and_forget TerminalPage::PasteFromClipboard(PasteFromClipboardEventArgs eventArgs)
+    {
+        const DataPackageView data = Clipboard::GetContent();
+
+        // This will switch the execution of the function to a background (not
+        // UI) thread. This is IMPORTANT, because the getting the clipboard data
+        // will crash on the UI thread, because the main thread is a STA.
+        co_await winrt::resume_background();
+
+        hstring text = L"";
+        if (data.Contains(StandardDataFormats::Text()))
+        {
+            text = co_await data.GetTextAsync();
+        }
+        eventArgs.HandleClipboardData(text);
+    }
+
+    // Method Description:
+    // - Copy text from the focused terminal to the Windows Clipboard
+    // Arguments:
+    // - trimTrailingWhitespace: enable removing any whitespace from copied selection
+    //    and get text to appear on separate lines.
+    void TerminalPage::_CopyText(const bool trimTrailingWhitespace)
+    {
+        const auto control = _GetFocusedControl();
+        control.CopySelectionToClipboard(trimTrailingWhitespace);
+    }
+
+    // Method Description:
+    // - Paste text from the Windows Clipboard to the focused terminal
+    void TerminalPage::_PasteText()
+    {
+        const auto control = _GetFocusedControl();
+        control.PasteTextFromClipboard();
+    }
+
+    // Method Description:
+    // - Removes the tab (both TerminalControl and XAML)
+    // Arguments:
+    // - tabViewItem: the TabViewItem in the TabView that is being removed.
+    void TerminalPage::_RemoveTabViewItem(const IInspectable& tabViewItem)
```

Comment:
Out of curiosity, most of the methods that moved to TerminalPage seem to have been re-ordered relative to one another. Any particular reason for this?

Reply:
Made some changes to the order of the functions in page. I moved all the functions with similar functionalities together and made every "group" consists with the old order